### ISSUE TITLE
Random crashes on Windows in Github actions

### ISSUE THREAD

**Issue by mmomtchev** [xpack-dev-tools/cmake-xpack#12]
I tend to run a very large amount of builds in Github Actions and I have seen on several occasions what appears to be a random crash.

It is quite rare, I have seen it only in Github Actions, always on Windows, always during the configure phase.

Here is one example:
https://github.com/mmomtchev/magickwand.js/actions/runs/11283317315/job/31382560438

I haven't been able to link this to any similar issues on the official CMake tracker.

Obviously, finding it will be very difficult - especially if it happens only in Github Actions. Maybe it is an interaction between the compiler runtime and the Github Actions Windows DLLs?

I am creating this issue to track the occurrences.

**Comment by ilg-ul** [comment]
> Maybe it is an interaction between the compiler runtime and the Github Actions Windows DLLs?

Maybe, I don't know. My first thought was for some resources that are exhausted, from memory to some internal handles or similar.

Could you find a trick to save the logs as artefacts, for later inspection?

**Comment by mmomtchev** [comment]
Yes, indeed, GHA runners do not have lots of memory. I will try to save the logs.

**Comment by ilg-ul** [comment]
Any updates on this?

**Comment by mmomtchev** [comment]
I have seen it a couple of times since, always in the same conditions - Github Actions, Windows platform, CMake during the configure phase and the only new information is that it always happens during the exact same configure test. Alas, I haven't been able to properly debug it, I am currently living in my van in freezing conditions because of the extortion, as I am been forced to accept a job without talking about an ongoing criminal affair that includes high-level corruption in the French and EU justice system. With limited amount of electricity, at the moment I am limited to classical paper-based research. If/when I get a new 12V computer, I will eventually move south during the spring, with the longer days and higher sun elevation, I will be able to resume my IT work.

**Comment by ilg-ul** [comment]
Ok, good luck!

Did you get my email about the new xPack site? I don't expect you to spend your limited amount of electricity on it, I just wanted to know if you received the email.

**Comment by mmomtchev** [comment]
Sure, I did.

**Comment by mmomtchev** [comment]
I just got this for the first time on a different CMake project (not the very complex ImageMagick) - https://github.com/mmomtchev/magickwand.js/actions/runs/14048759584/job/39335108019

This time it is `libjpeg-turbo` which is not a huge CMake project. Everything else stays the same - configure phase, Windows, GHA.

The error - 3221225477 - is basically a segmentation fault - https://trycatchdebug.net/news/1252241/debugging-c-mystery-3221225477-return-value (alas, there are some very annoying ads).

**Comment by mmomtchev** [comment]
GHA run on virtual machines with very few cores and I have seen synchronisation problems that manifest (almost) exclusively in GHA.

**Comment by ilg-ul** [comment]
In my opinion, your workflows are quite complex, perhaps too complex for the resources you have.
